zipline/utils/cli.py

```diff
@@ -1,285 +1,118 @@
-    # Read in a CSV file, if applicable
-    if asset_metadata_path is not None:
-        if os.path.isfile(asset_metadata_path):
-            asset_metadata = pd.read_csv(asset_metadata_path,
-                                         index_col=asset_identifier)
-
-    source_arg = kwargs['source']
-    source_time_column = kwargs['source_time_column']
-
-    if source_arg is None:
-        raise NoSourceError()
-
-    elif source_arg == 'yahoo':
-        source = zipline.data.load_bars_from_yahoo(
-            stocks=symbols, start=start, end=end)
-
-    elif os.path.isfile(source_arg):
-        source = zipline.data.load_prices_from_csv(
-            filepath=source_arg,
-            identifier_col=source_time_column
-        )
+    def __init__(self, tz=None, as_timestamp=False):
+        super(Date, self).__init__(tz=tz)
+        self.as_timestamp = as_timestamp
 
-    elif os.path.isdir(source_arg):
-        source = zipline.data.load_prices_from_csv_folder(
-            folderpath=source_arg,
-            identifier_col=source_time_column
-        )
+    def parser(self, value):
+        ts = super(Date, self).parser(value)
+        return ts.normalize() if self.as_timestamp else ts.date()
 
-    else:
-        raise NotImplementedError(
-            'Source %s not implemented.' % kwargs['source'])
 
-    algo_text = kwargs.get('algo_text', None)
-    if algo_text is None:
-        # Expect algofile to be set
-        algo_fname = kwargs['algofile']
-        with open(algo_fname, 'r') as fd:
-            algo_text = fd.read()
+class Time(_DatetimeParam):
+    """A click parameter that parses the value into timetime.time objects.
 
-    if print_algo:
-        if PYGMENTS:
-            highlight(algo_text, PythonLexer(), TerminalFormatter(),
-                      outfile=sys.stdout)
-        else:
-            print_(algo_text)
-
-    algo = zipline.TradingAlgorithm(script=algo_text,
-                                    namespace=kwargs.get('namespace', {}),
-                                    capital_base=float(kwargs['capital_base']),
-                                    algo_filename=kwargs.get('algofile'),
-                                    equities_metadata=asset_metadata,
-                                    start=start,
-                                    end=end)
-
-    perf = algo.run(source, overwrite_sim_params=overwrite_sim_params)
-
-    output_fname = kwargs.get('output', None)
-    if output_fname is not None:
-        perf.to_pickle(output_fname)
-
-    return perf
+    Parameters
+    ----------
+    tz : timezone-coercable, optional
+        The timezone to parse the string as.
+        By default the timezone will be infered from the string or naiive.
+    """
+    def parser(self, value):
+        return super(Time, self).parser(value).time()
 
 
-def maybe_show_progress(it, show_progress, **kwargs):
-    """Optionally show a progress bar for the given iterator.
+class Timedelta(_DatetimeParam):
```

Comment:
I don't think I ended up using it, We might want to add an `--older-than` flag to `clean` which would want this